source/geometries/GenericWLSFiber.h

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------------
+//  nexus | GenericWLSFiber.h
+//
+//  Geometry of a configurable wave-length shifting optical fiber.
+//
+//  The NEXT Collaboration
+// -----------------------------------------------------------------------------
+
+#ifndef GENERIC_WLS_FIBER_H
+#define GENERIC_WLS_FIBER_H
+
+#include "GeometryBase.h"
+#include <G4MaterialPropertyVector.hh>
+
+class G4Material;
+class G4GenericMessenger;
+class G4MaterialPropertiesTable;
+
+namespace nexus {
+
+  class GenericWLSFiber: public GeometryBase
+  {
+
+  public:
+    // Constructor for a generic wave-length shifting optical fiber
+    GenericWLSFiber(G4String    name,
+                    G4String    shape,            // "round"  or "square"
```

Comment:
This should be an _enum_ or a _bool_ (there are only two options) instead of a _string_.